### PR TITLE
`step ca renew` in offline mode should not require `--ca-url` and `--root` flags.

### LINKED ISSUE #627: Offline "step ca renew" insists on --ca-url and --root parameters
### Subject of the issue
It is irritating that the command
```step ca renew renewed.crt renewed.key --offline --ca-config ./config/ca.json ``` requires the --ca-url and --root parameters.
The documentation at https://smallstep.com/docs/step-cli/reference/ca/renew has a sample invocation that does not work at all for me:
```step ca renew --offline internal.crt internal.key```

### Your environment
* OS - Linux (Ubuntu 20.04)
* Version - master (84e2b87739a581ea0e83efa0861fc671a9d51654)

### Steps to reproduce

- Create an offline CA with CA config at config/ca.json (not the default path)
- Issue a certificate with ```step ca certificate c c.crt c.key --offline --ca-config config/ca.json```
- Renew that certificate with ```step ca renew c.crt c.key --offline --ca-config config/ca.json```

### Expected behaviour
The certificate can be renewed with the command line above and/or a correct command line for this command is supplied at https://smallstep.com/docs/step-cli/reference/ca/renew

### Actual behaviour
The command aborts with 
```'step ca renew' requires the '--ca-url' flag```
When a dummy URL is provided, the command 
```step ca renew c.crt c.key --offline --ca-config config/ca.json --ca-url https://dummy.url``` 

complains:

```open /home/user/.step/certs/root_ca.crt failed: no such file or directory```
The "root" property in config/ca.json is not honored.

When giving the --root and --ca-url options, the certificate is renewed as expected.

### PR DESCRIPTION
**Pull request by henschkowski**
Fixes #627.

I'm new to Go - this is my attempt to fix the issue.

💔Thank you!

### CODE REVIEW

**Comment by dopey** [comment]
@henschkowski, thanks for the PR! Would you mind signing our CLA please?

**Comment by henschkowski** [comment]
Signed it .

Max ***@***.***> schrieb am Mi., 9. Feb. 2022, 20:57:

File: flags/flags.go [hunk 1]
```diff
@@ -445,7 +445,8 @@ func ParseTemplateData(ctx *cli.Context) (json.RawMessage, error) {
 //  - Error if the URL scheme is not implicitly or explicitly 'https'.
 func ParseCaURL(ctx *cli.Context) (string, error) {
 	caURL := ctx.String("ca-url")
-	if caURL == "" {
+	if caURL == "" && 
+           !ctx.IsSet("offline") {
```
Comment (maraino) [inline flags/flags.go:449]: I would use `!ctx.Bool("offline")` because --offline=false is a valid option too.

**Review by maraino — CHANGES_REQUESTED** [review]
Hi @henschkowski, thanks for the PR, but can you change the condition.

**Comment by dopey** [comment]
Pulled this into another PR to fix the conflict which is from something else we recently merged.

**Pull request closed without merging**